devlib/instrument/monsoon.py

```diff
@@ -0,0 +1,170 @@
+import devlib
+import json
+import os
+import os.path
+import psutil
+import time
+import logging
+from subprocess import Popen, STDOUT
+from threading import Timer
+import pandas as pd
+import StringIO
+from collections import namedtuple
+from subprocess import Popen, PIPE, STDOUT
+from devlib.instrument import Instrument, CONTINUOUS, MeasurementsCsv
+from devlib.exception import HostError
+from devlib.utils.misc import which
+
+class MonsoonInstrument(Instrument):
+    MONSOON_BIN = "./tools/scripts/monsoon.par"
+
+    def log_error(self, errmsg, exception=0):
+	print "ERROR: " + errmsg
+	self.logger.error(errmsg)
+	if exception == 1:
+		raise(Exception(errmsg))
+
+    def log_info(self, msg):
+	print "INFO: " + msg
+	self.logger.info(msg)
+
+    def __init__(self, target, device_entry=None):
+	super(MonsoonInstrument, self).__init__(target)
+
+	if not os.path.exists(MonsoonInstrument.MONSOON_BIN):
+		self.log_error("monsoon.par not found", exception=1)
+
+	if device_entry != None and not os.path.exists(device_entry):
+		self.log_error(str(device_entry) + " doesn't exist", exception=1)
+
+	self.device_entry = device_entry
+	self.process = None
+
+	self.log_info('Found monsoon.par')
```

Comment:
Do not report "it's working" type of comments, let's log only useful information and/or warnings/errors.

In devlib modules are "as much quite as possible". ;-)